File: 1). DSA/1). Data Structures/01). Array/C++/_042)_Find_Equilibrium_Point_in_an_Array.cpp
#include<iostream>
using namespace std;

bool checkEquilibrium(int arr[], int n)
{
	for(int i  = 0; i < n; i++)
	{
		int l_sum = 0, r_sum = 0;
		for(int j = 0; j < i; j++)
			l_sum += arr[j];
		for(int j = i + 1; j < n; j++)
			r_sum += arr[j];
		if(l_sum == r_sum)
			return true;
	}
	return false;
}



int main()
{
    int arr[] = {3, 4, 8, -9, 20, 6}, n = 6;
    cout<<checkEquilibrium(arr, n)? "true" : "false";
    cout<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_042)_Find_Equilibrium_Point_in_an_Array.dart
bool checkEquilibrium(List<int> arr) {
  for (int i = 0; i < arr.length; i++) {
    int lSum = 0, rSum = 0;
    for (int j = 0; j < i; j++) lSum += arr[j];

    for (int j = i + 1; j < arr.length; j++) rSum += arr[j];

    if (lSum == rSum) return true;
  }
  return false;
}

void main() {
  var arr = [3, 4, 8, -9, 20, 6];
  print('Equilibrium = ${checkEquilibrium(arr)}');
}
